frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -1677,6 +1683,73 @@ class Worksheet extends React.Component {
             },
         );
     };
+
+    showBundleContent = () => {
+        this.setState({ openedDialog: DIALOG_TYPES.OPEN_CREATE_CONTENT });
+    };
+    showBundleContentCallback = (path) => () => {
+        // Default path for content block is '/'
+        path = path || '/';
+        let validBundles = [];
+        let showContentCounts = 0;
+        let tableIDs = Object.keys(this.showContentCallbacks).sort();
+        tableIDs.forEach((tableID) => {
+            let showContentBundleCallback = this.showContentCallbacks[tableID];
+            let bundlesChecked = showContentBundleCallback();
+            bundlesChecked.forEach((bundle) => {
+                if (bundle.name === '<invalid>') {
+                    return;
+                }
+                validBundles.push(bundle.rawIndex);
+                showContentCounts += 1;
+            });
+        });
+
+        let newSource = [...this.state.ws.info.source];
+
+        validBundles.forEach((index, i) => {
+            const items = ['% display contents ' + path, newSource[index], ''];
+            // For each checked bundle, we are adding 3 new lines to the source right after each checked bundle
+            newSource = [
+                ...newSource.slice(0, index + 3 * i + 1),
```

Comment:
Can we instead use `newSource = [...newSource].splice(...)` to insert the 3 new lines? it might be a bit clearer: https://stackoverflow.com/a/586189

Reply:
updated